### DISCUSSION TITLE
auth: Magic Link [wishlist]

### DISCUSSION THREAD

**Discussion by gpalazuelosg** [golang-app/ecommerce#58] [Ideas]
A user should be able to logic using magic links (no password needed).

This is a nice to have.

**Comment by bkielbasa** [comment]
Hey, thanks for your suggestion. How do you imagine it to work?

**Reply by gpalazuelosg** [reply]
Hi Bartek.

- A customer user arrives to the website, he decides to login into our site.
- To login, among the potential others options, he chooses the option "Magic Email Link".
- He provides his/her email.
- Screen cheers with message "Please check your email for instructions to login".
- A button/link suggest user "Other login options" (if other options exist).

The case is for users like [my father, any other] which hates capturing password because he forgets about it, they hates remembering multiple passwords in multiple sites.

**Reply by bkielbasa** [reply]
I see. I have some security concerns about the Magic link. When someone gets access to the email account, automatically gets access to the website as well. Personally, I'd prefer to implement other passwordless versions like Biometrics or Possession factors. Using a YubiKey sounds like a good option as well.

I have to think about it and maybe wait for a voice from other people.

Thanks for your suggestion, it sounds quite interesting, tho!

**Reply by bkielbasa** [reply]
There's one more option - integrate with oauth0 and forget about all of these things here: https://auth0.com/. It has a free account with up to 7k unique users that should fit perfectly into this project.
